config/provider.go

```diff
@@ -0,0 +1,232 @@
+	"aws_ec2_transit_gateway_route$",
+	"aws_ec2_transit_gateway_route_table$",
+	"aws_ec2_transit_gateway_route_table_association$",
+	"aws_ec2_transit_gateway_vpc_attachment$",
+	"aws_ec2_transit_gateway_vpc_attachment_accepter$",
+	"aws_ec2_transit_gateway_route_table_propagation$",
+
+	// IAM
+	"aws_iam_access_key$",
+	"aws_iam_group$",
+	"aws_iam_group_policy$",
+	"aws_iam_group_policy_attachment$",
+	"aws_iam_instance_profile$",
+	"aws_iam_policy$",
+	"aws_iam_policy_attachment$",
+	"aws_iam_role$",
+	"aws_iam_role_policy$",
+	"aws_iam_role_policy_attachment$",
+	"aws_iam_user$",
+	"aws_iam_user_group_membership$",
+	"aws_iam_user_policy$",
+	"aws_iam_user_policy_attachment$",
+
+	// EKS
+	"aws_eks_addon$",
+	"aws_eks_cluster$",
+	"aws_eks_fargate_profile$",
+	"aws_eks_node_group$",
+	"aws_eks_identity_provider_config$",
+
+	// KMS
+	"aws_kms_key$",
+
+	// EBS
+	"aws_ebs_volume$",
+}
+
+// These resources cannot be generated because of their suffixes colliding with
+// kubebuilder-accepted type suffixes.
+var skipList = []string{
+	"aws_config_configuration_recorder_status$",
+	"aws_vpc_peering_connection_accepter$",
+	"aws_elasticache_user_group$",
+	"aws_waf_rule_group$",
+	"aws_wafregional_rule_group$",
+	"aws_shield_protection_group$",
+	"aws_route53_resolver_firewall_rule_group$",
+	"aws_kinesis_analytics_application$",
+}
+
+// GetProvider returns provider configuration
+func GetProvider(tfProvider *schema.Provider) *tjconfig.Provider {
+	pc := tjconfig.NewProvider(
+		tfProvider.ResourcesMap, resourcePrefix, "github.com/crossplane-contrib/provider-tf-aws",
+		tjconfig.WithIncludeList(includedResources),
+		tjconfig.WithSkipList(skipList),
+		tjconfig.WithDefaultResourceFn(DefaultResource),
+	)
+
+	for _, configure := range []func(provider *tjconfig.Provider){
+		autoscaling.Configure,
+		ebs.Configure,
+		ec2.Configure,
+		ecr.Configure,
+		ecrpublic.Configure,
+		ecs.Configure,
+		eks.Configure,
+		elasticache.Configure,
+		elasticloadbalancing.Configure,
+		iam.Configure,
+		kms.Configure,
+		rds.Configure,
+		s3.Configure,
+	} {
+		configure(pc)
+	}
+
+	pc.ConfigureResources()
+	return pc
+}
+
+// DefaultResource returns a base resource configuration for AWS resources.
+func DefaultResource(name string, terraformResource *schema.Resource) *tjconfig.Resource { // nolint:gocyclo
+	// NOTE(muvaf): gocyclo is disabled because this function is mostly for if
+	// conditions that are easy to grasp.
+	r := tjconfig.DefaultResource(name, terraformResource)
+
+	switch {
+	case strings.HasPrefix(name, "aws_security_group") ||
+		strings.HasPrefix(name, "aws_subnet") ||
+		strings.HasPrefix(name, "aws_network") ||
+		strings.HasPrefix(name, "aws_eip") ||
+		strings.HasPrefix(name, "aws_launch_template") ||
+		strings.HasPrefix(name, "aws_route") ||
+		strings.HasPrefix(name, "aws_instance") ||
+		strings.HasPrefix(name, "aws_vpc"):
+		r.ShortGroup = "ec2"
+	case r.ShortGroup == "vpc":
+		r.ShortGroup = "ec2"
+	case name == "aws_lb_":
```

Comment:
```suggestion
	case strings.HasPrefix(name, "aws_lb"):
```

Reply:
I added a map with regexes and it covers both.